lib/diagnostic.js

```diff
@@ -8,39 +8,60 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-const min = function min() {
-
+const min = function min(prev, curr) {
+  if (curr < prev) {
+    return curr;
+  }
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-const sum = function sum() {
-
-};
+const sum = function sum(prev, curr) {
+    return prev + curr;
+  };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 const any = function any(array, callback) {
-
-};
+  let result = false;
+     for (let i = 0; i < array.length; i++) {
+       if (!callback(array[i])) {
+         result = true;
+       }
+     }
+     return result;
+   };
 
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 const select = function select(array, callback) {
-
-};
+    let result = false;
+    for (let i = 0; i < array.length; i++) {
+      if (callback(array[i]) === true) {
+        let result = [];
+      result.push(callback(array[i]));
+    }
+     }
+     return result;
+   };
```

Comment:
So close here! If result is your accumulator and you want to be pushing elements into it, it'd be best to initialize it as an empty array. 

```
const select = function select(array, callback) {
  let result = [];
  array.forEach(function(currentValue, index, array) {
    if (callback(currentValue, index, array)) {
      result.push(currentValue);
    }
  });

  return result;
};
```